var/spack/repos/builtin/packages/intel/package.py

```diff
@@ -0,0 +1,113 @@
+from spack import *
+import os
+
+class Intel(Package):
+    """Intel Compilers.
+
+    Note: You will have to add add the downolad file to a
+    mirror so that Spack can find it. For instructions on how to set up a
+    mirror, see http://software.llnl.gov/spack/mirrors.html"""
+
+    homepage = "http://www.intel.com"
+
+    # TODO: can also try the online installer (will download files on demand)
+    version('composer.2016.2', '1133fb831312eb519f7da897fec223fa',
+            url="file://%s/parallel_studio_xe_2016_composer_edition_update2.tgz" % os.getcwd())
+    version('professional.2016.2', '70be832f2d34c9bf596a5e99d5f2d832',
+            url="file://%s/parallel_studio_xe_2016_update2.tgz" % os.getcwd())
+    version('cluster.2016.2', '70be832f2d34c9bf596a5e99d5f2d832',
+            url="file://%s/parallel_studio_xe_2016_update2.tgz" % os.getcwd())
+
+    variant('rpath', default=True, description="Add rpath to .cfg files")
+    variant('all', default=False, description="Install all files associated with the requested edition")
+    variant('mpi', default=True, description="Install the Intel MPI library and ITAC tool")
+    variant('mkl', default=True, description="Install the Intel MKL library")
+    variant('daal', default=True, description="Install the Intel DAAL libraries")
+    variant('ipp', default=True, description="Install the Intel IPP libraries")
+    variant('tools', default=True, description="Install the Intel Advisor, VTune Amplifier, and Inspector tools")
+
+    provides('mpi', when='@cluster:+mpi')
+    provides('mkl', when='+mkl')
```

Comment:
I think that sounds reasonable, but how well would that work since the library names are different? I suppose we could create symlinks with the standard libblas and liblapack names pointing to the appropriate mkl libraries. We would also have to create some symlinks to the have {prefix}/lib directory point to MKL's lib directory. The Intel package is a big mess, eh?!